### PR TITLE
feat: percentile disc

### PR DESCRIPTION
**Pull request by pomykalakyle**
## Add `percentile_disc` aggregate function

Part of #309

### Summary

Implements the `percentile_disc` (discrete percentile) aggregate function for Spark SQL compatibility. Unlike `percentile_cont` which interpolates between values, `percentile_disc` returns an actual value from the dataset at the requested percentile.

### Changes

**New files:**
- `crates/sail-function/src/aggregate/percentile_disc.rs` - UDAF definition with signature and return types
- `crates/sail-function/src/aggregate/percentile_disc_groups.rs` - `GroupsAccumulator` implementation for efficient GROUP BY queries
- `crates/sail-function/src/aggregate/utils.rs` - Shared utilities including `calculate_percentile_disc`

**Modified:**
- `crates/sail-plan/src/function/aggregate.rs` - Wire up `percentile_disc_expr` to the function registry
- `python/pysail/data/compatibility/functions/aggregate.json` - Mark as supported
- `crates/sail-spark-connect/tests/gold_data/function/agg.json` - Updated gold test data

**Tests:**
- `python/pysail/tests/spark/test_percentile.py` - Added tests for basic usage, GROUP BY, and edge cases (0th/100th percentiles)

### Notes

- The implementation follows DataFusion's `percentile_cont` patterns, including the chunked bitmap optimization for null handling in the groups accumulator

### CODE REVIEW

File: crates/sail-function/src/aggregate/percentile_disc_groups.rs [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+
+        Ok(vec![Arc::new(converted_list_array)])
+    }
+
+    fn supports_convert_to_state(&self) -> bool {
+        true
+    }
+
+    fn size(&self) -> usize {
+        self.group_values
```
Comment (linhr) [inline crates/sail-function/src/aggregate/percentile_disc_groups.rs]: Do we need to include `size_of_val(self)`?

**Review by linhr — COMMENTED** [review]
This looks great! 🚀

Could you update `crates/sail-execution/src/codec.rs` to register the custom aggregate function so that it can be recognized in distributed query execution? You can take a look at how other UDAFs are added there. (We hope to improve our test suites soon so that the missing codec issue can be identified via tests.)

File: crates/sail-function/src/aggregate/percentile_disc_groups.rs (hunk 1 above)
Comment (pomykalakyle) [reply crates/sail-function/src/aggregate/percentile_disc_groups.rs]: I was following what datafusion was doing but this is a good catch - maybe this was overlooked in datafusion?

File: python/pysail/tests/spark/test_percentile.py [hunk 2]
```diff
@@ -20,3 +20,74 @@ def test_percentile_cont(spark):
...
+        SELECT
+            percentile_disc(0.0) WITHIN GROUP (ORDER BY val) AS p0,
+            percentile_disc(1.0) WITHIN GROUP (ORDER BY val) AS p100
+        FROM test_disc_edges
+    """).toPandas()
+
+    # 0th percentile = min = 5
+    # 100th percentile = max = 20
+    expected = pd.DataFrame({"p0": [5], "p100": [20]})
+    pd.testing.assert_frame_equal(actual, expected)
```
[the remark this answers is not in the document]
Comment (pomykalakyle) [reply python/pysail/tests/spark/test_percentile.py:93]: added a test for this

**Comment by pomykalakyle** [comment]
> This looks great! 🚀
> 
> Could you update `crates/sail-execution/src/codec.rs` to register the custom aggregate function so that it can be recognized in distributed query execution? You can take a look at how other UDAFs are added there. (We hope to improve our test suites soon so that the missing codec issue can be identified via tests.)

added

File: crates/sail-function/src/aggregate/utils.rs [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+        Err(DataFusionError::Internal(
+            "Didn't expect ColumnarValue::Array".to_string(),
+        ))
+    }
+}
+
+pub fn validate_percentile(expr: &Arc<dyn PhysicalExpr>) -> Result<f64> {
+    let scalar_value = get_scalar_value(expr).map_err(|_e| {
+        DataFusionError::Plan(
+            "Percentile value for 'PERCENTILE_DISC' must be a literal".to_string(),
```
[the remark this answers is not in the document]
Comment (pomykalakyle) [reply crates/sail-function/src/aggregate/utils.rs:36]: I don't think this is needed.

File: crates/sail-function/src/aggregate/utils.rs [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+        )
+    })?;
+
+    let percentile = match scalar_value {
+        ScalarValue::Float32(Some(value)) => value as f64,
+        ScalarValue::Float64(Some(value)) => value,
+        sv => {
+            return Err(DataFusionError::Plan(format!(
+                "Percentile value for 'PERCENTILE_DISC' must be Float32 or Float64 literal (got data type {})",
```
[the remark this answers is not in the document]
Comment (pomykalakyle) [reply crates/sail-function/src/aggregate/utils.rs:45]: I don't think this is needed.

**Review by linhr — APPROVED** [review]
Thanks for the contribution!
